packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,332 @@
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import { Construct, IResource, Lazy, Resource, Stack, Token, Duration } from '@aws-cdk/core';
+import { CfnDistribution } from './cloudfront.generated';
+import { Origin } from './origin';
+import { CacheBehavior } from './private/cache-behavior';
+
 /**
  * Interface for CloudFront distributions
  */
-export interface IDistribution {
+export interface IDistribution extends IResource {
+  /**
+   * The domain name of the Distribution, such as d111111abcdef8.cloudfront.net.
+   *
+   * @attribute
+   */
+  readonly domainName: string;
+
+  /**
+   * The distribution ID for this distribution.
+   *
+   * @attribute
+   */
+  readonly distributionId: string;
+}
+
+/**
+ * Attributes used to import a Distribution.
+ *
+ * @experimental
+ */
+export interface DistributionAttributes {
   /**
-   * The domain name of the distribution
+   * The generated domain name of the Distribution, such as d111111abcdef8.cloudfront.net.
+   *
+   * @attribute
    */
   readonly domainName: string;
 
   /**
    * The distribution ID for this distribution.
+   *
+   * @attribute
    */
   readonly distributionId: string;
+}
+
+/**
+ * Properties for a Distribution
+ *
+ * @experimental
+ */
+export interface DistributionProps {
+  /**
+   * The default behavior for the distribution.
+   */
+  readonly defaultBehavior: BehaviorOptions;
+
+  /**
+   * Additional behaviors for the distribution, mapped by the pathPattern that specifies which requests to apply the behavior to.
+   *
+   * @default no additional behaviors are added.
+   */
+  readonly additionalBehaviors?: Record<string, BehaviorOptions>;
+
+  /**
+   * A certificate to associate with the distribution. The certificate must be located in N. Virginia (us-east-1).
+   *
+   * @default the CloudFront wildcard certificate (*.cloudfront.net) will be used.
```

Comment:
If the default is descriptive (and not a specific value), then start with a `-` because the IDE expects a value as the first element in the default description:


```
@default - the CloudFront wildcard ....
```